assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/components/table/footerRowsGenerator.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { DownCircleFilled, UpCircleFilled } from '@ant-design/icons';
+import { __ } from '@wordpress/i18n';
+
+import { Cell } from '@appLayout/espressoTable';
+import { EspressoButton } from '@application/ui/input';
+import { parsedAmount } from '@appServices/utilities/money';
+import { TicketPriceField } from '../../fields';
+import { useDataState } from '../../data';
+import { useMoneyDisplay } from '@appServices/utilities/money';
+
+const footerRowsGenerator = () => {
+	const { reverseCalculate, toggleCalcDir } = useDataState();
+	const { formatAmount } = useMoneyDisplay();
+	const calcDirIcon = reverseCalculate ? <UpCircleFilled /> : <DownCircleFilled />;
```

Comment:
We need to find a way to avoid these hooks in these generators. Although they are not a problem here and in `EntityList` generators because the order won't change on rerender, but still  it's not a good approach.

Reply:
agree, applied temp `useMemo` in this case until we will decide a better global approach

Reply:
those could be passed in from the parent component or via a data container component that wraps the generator

Reply:
lifted up: https://github.com/eventespresso/event-espresso-core/pull/2651/commits/e9443e0107e92645e66c6621b5c1bf41c2e00117
***
https://www.youtube.com/watch?v=pT_Y-eodTv4